pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -211,6 +211,16 @@ func ApplySearchHeadCluster(ctx context.Context, client splcommon.ControllerClie
 		cr.Status.AdminPasswordChangedSecrets = make(map[string]bool)
 		cr.Status.NamespaceSecretResourceVersion = namespaceScopedSecret.ObjectMeta.ResourceVersion
 
+		// Add a splunk operator telemetry app
+		if !cr.Status.TelAppInstalled && !cr.Spec.Mock {
```

Comment:
For some instances ephemeral storage is common, so, we should cover that scenario as well. Otherwise, we will lose the app across the resets. Indexers are the best examples. I think, if we use the bundle push logic for IDXC, then it is implicitly covered(assuming that the Cluster Master always uses the persistent storage)?

Reply:
Understood Subba. I have to introduce logic here to check whether the folder exists if its ephemeral storage.